File: app/src/main/java/org/schulcloud/mobile/data/model/Submission.java
package org.schulcloud.mobile.data.model;

import io.realm.RealmList;
import io.realm.RealmModel;
import io.realm.annotations.PrimaryKey;
import io.realm.annotations.RealmClass;

@RealmClass
public class Submission implements RealmModel {
    @PrimaryKey
    public String _id;
    public String schoolId;
    public String homeworkId;
    public String studentId;
    public String comment;
    public String createdAt;

    public Integer grade;
    public String gradeComment;

    public RealmList<Comment> comments;
}

File: app/src/main/java/org/schulcloud/mobile/models/homework/submission/Submission.kt
package org.schulcloud.mobile.models.homework.submission

import com.google.gson.annotations.SerializedName
import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import org.schulcloud.mobile.models.base.HasId

open class Submission : RealmObject(), HasId {

    @PrimaryKey
    @SerializedName("_id")
    override var id: String = ""

    var homeworkId: String? = null
    var studentId: String? = null
    var comment: String? = null
    var createdAt: String? = null

    var grade: Int? = null
    var gradeComment: String? = null

    var comments: RealmList<Comment>? = null

}
